src/base/static/libs/maps/index.js

```diff
@@ -225,75 +222,66 @@ class MainMap {
       "USER",
       "map",
       "geolocate",
-      this.map.getBBoxString(),
-      this.map.getZoom(),
+      this._map.getBBoxString(),
+      this._map.getZoom(),
     );
     this.geolocate();
   }
 
   geolocate() {
-    this.map.locate();
-  }
-
-  addLayerView(collectionId) {
-    return model => {
-      this.layerViews[collectionId][model.cid] = new LayerView({
-        model: model,
-        router: this.router,
-        map: this.map,
-        layerGroup: this.layers[collectionId],
-        placeTypes: this.options.placeTypes,
-        // to access the filter
-        mapView: this,
-      });
-    };
+    // TODO
+    this._map.locate();
   }
 
-  removeLayerView(collectionId) {
-    return model => {
-      // remove map-bound events for this layer view
-      this.map.off(
-        "zoomend",
-        this.layerViews[collectionId][model.cid].updateLayer,
-        this.layerViews[collectionId][model.cid],
-      );
-      this.map.off(
-        "move",
-        this.layerViews[collectionId][model.cid].throttledRender,
-        this.layerViews[collectionId][model.cid],
-      );
-      this.map.off(
-        "zoomend",
-        this.layerViews[collectionId][model.cid].render,
-        this.layerViews[collectionId][model.cid],
+  updatePlaceCollectionLayer(collectionId, modelId) {
+    this._map
+      .getSource(collectionId)
+      .setData(
+        createGeoJSONFromCollection(this._places[collectionId], modelId),
       );
+  }
 
-      this.layerViews[collectionId][model.cid].remove();
-      delete this.layerViews[collectionId][model.cid];
+  addPlaceCollectionLayer(collectionId) {
+    this._map.createGeoJSONLayer({
+      id: collectionId,
+      rules: this._layers.find(layer => layer.id === collectionId).rules,
+      source: createGeoJSONFromCollection(this._places[collectionId]),
+    });
```

Comment:
Can we add a comment here to clarify that this object here is a Layer instance? I want to be explicit about the data structures that we are using.

Eventually, I think we will be passing the Layer itself to `addPlaceCollectionLayer` as well as in the payload of the event that triggered it, but it probably isn't worth doing now.

Reply:
I suppose our Layers are only business class at the moment :)

Reply:
But yeah, I'd say let's leave it the way it is for now, and we can upgrade them to First soon.